imagetest/test_suites/ssh/image_ssh_test.go

```diff
@@ -67,13 +71,37 @@ func createSession(user, host string, pembytes []byte) (*ssh.Client, *ssh.Sessio
 
 	client, err := ssh.Dial("tcp", host, sshConfig)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
+	return client, nil
+}
 
+// checkLocalUser test that the user account exists in /etc/passwd
+func checkLocalUser(client *ssh.Client, user string) error {
 	session, err := client.NewSession()
 	if err != nil {
 		client.Close()
-		return nil, nil, err
+		return err
+	}
+	defer session.Close()
+	grepPasswdCmd := fmt.Sprintf("grep %s: /etc/passwd", user)
+	if err := session.Run(grepPasswdCmd); err != nil {
+		return err
+	}
+	return nil
```

Comment:
This looks like it assumed success if there is no error. 

Is it possible to ensure the command succeeded in such a way that the output is also verifed. 
Ex: How would it know the difference between running and successfully returning with no results vs getting the expected result?

Reply:
Check the manual of grep command:
```
Normally, the exit status is 0 if selected lines are found and 1 otherwise. But the exit status is 2 if an error occurred, unless the -q or --quiet or --silent option is used and a selected line is found. 
```
So, i thought it would be fine to just use error code 0 to indicate that found the line matched.

Reply:
https://linux.die.net/man/1/grep